### ISSUE TITLE
[FEATURE] Add ability to set multiple databricks_git_credential resources

### ISSUE THREAD

**Issue by mariohoxha42** [databricks/terraform-provider-databricks#3430]
### Use-cases

The use case for this feature would be to allow for the creation of more than one `databricks_git_credential` for a workspace when created via Terraform as this is required for CI/CD processes when enabling jobs to run as a particular service principal. The reason why a job would need to run as a service principal is to allow for continuity in a workspace when people leave a workspace and take with them their PAT that is used to run a job. This causes failures for teams and downtime for critical workflows that Databricks aims to support for their clients' organizations. 

### Attempted Solutions
The best solution I can find is the following: https://github.com/databricks/terraform-provider-databricks/issues/2707#issuecomment-1748629459

While this can work with one service principal and one `databricks_git_credential`, it is not a very clean way to do it. For workspaces that support multiple teams with various needs and access to different repos, scaling this to multiple credentials for multiple teams get very unpleasant quickly with a multiplying number of providers. 

The only other way to do it [as per the service principal for CI/CD docs](https://docs.databricks.com/en/dev-tools/ci-cd/ci-cd-sp.html#add-git-provider-credentials-to-a-databricks-workspace) is to use the Databricks CLI to set the `databricks_git_credential` which defeats the purpose of a unified CI/CD process via Terraform and involves manual intervention

### Proposal
I propose that a new optional parameter is added to `databricks_git_credential` that takes in `application_id` from a service principal. This is to support the setting the credential for a given Service Principal and allow for users to centrally define their service principal, secrets, and git credentials in one place, and later use that service principal in their `databricks_job`'s to run as a given service principal

### References
#2707

**Comment by alexott** [comment]
It's possible to do this by using a dedicated provider instance for that service principal and then creating `databricks_git_credential` with that provider instance.

**Comment by mariohoxha42** [comment]
Yes, while that works, it does make the repo we use to manage our terraform more complex and harder to understand by adding a new `databricks_service_principal_secret` object, a new provider block, all for the express purpose of that new `databricks_git_credential`. It'd be very similar to the solution in other issue I referenced. 

As we already have two additional providers setup for account-level and workspace-level, I was looking for simpler functionality to that existing `databricks_git_credential` resource that would make setting the credentials not as complicated. Adding `n` dedicated providers for `n` service principals that we want to do this for would just add more complexity as we are already navigating two databricks providers as it is.

**Comment by alexott** [comment]
That's what is available now - there is no API to set a git credential for another user/SP, so Terraform can't do anything...

**Comment by alexott** [comment]
supported now

**Comment by brandocomando** [comment]
Can you give an example? I dont see any documentation on how to use this resource to create a git credential on behalf of a service principal?

**Comment by alexott** [comment]
https://github.com/alexott/terraform-playground/tree/main/ado-adb-sp-git, https://github.com/alexott/terraform-playground/tree/main/git-credential-for-sp
